### PR TITLE
Add `NumberInterval::union()` tests and fix covering closed intervals with open interval with same start and end

### PR DESCRIPTION
**Pull request by simPod**

### CODE REVIEW

File: tests/Number/NumberIntervalTest.php [hunk 1]
```diff
@@ -117,6 +118,32 @@ public function testIntersectInterval($origin_interval, $compare_interval, $expe
...
+    public function providerUnionInterval(): Generator
+    {
+        yield 'Closed with closed' => ['[5,10]', '[1,7]', '[1,10]'];
+        yield 'Open with closed' => ['(5,8)', '[5,8]', '[5,8]'];
+        yield 'Closed with open' => ['[5,8]', '(5,8)', '[5,8]'];
+        yield 'Not intersecting' => ['(1,2]', '[4,7)', null];
+        yield 'Open start with open end' => ['(1,4]', '[4,7)', '(1,7)'];
+        yield 'Open start with closed end' => ['(5,8]', '[6,9]', '(5,9]'];
+        yield 'Closed start with open end' => ['[5,8]', '(6,9)', '[5,9)'];
+    }
```
Comment (peter-gribanov) [inline tests/Number/NumberIntervalTest.php:136]: Please add test case:
```php
yield 'Open with open' => ['(5,8)', '(5,8)', '(5,8)'];
```

**Review by peter-gribanov — CHANGES_REQUESTED** [review]

File: tests/Number/NumberIntervalTest.php [hunk 2]
```diff
@@ -117,6 +118,32 @@ public function testIntersectInterval($origin_interval, $compare_interval, $expe
         $this->assertEquals($expected_interval, $origin_interval->intersection($compare_interval));
     }
 
+    /**
+     * @return Generator<array-key, array{string, string, string|null}>
+     */
+    public function providerUnionInterval(): Generator
+    {
+        yield 'Closed with closed' => ['[5,10]', '[1,7]', '[1,10]'];
+        yield 'Open with closed' => ['(5,8)', '[5,8]', '[5,8]'];
```
Comment (simPod) [inline tests/Number/NumberIntervalTest.php:130]:
```suggestion
        yield 'Open with closed' => ['(5,8)', '[5,8]', '[5,8]'];
        yield 'Open with open' => ['(5,8)', '(5,8)', '(5,8)'];
```

**Comment by simPod** [comment]
@peter-gribanov btw try to re-enable CI action at https://github.com/gpslab/interval/actions

**Review by peter-gribanov — APPROVED** [review]

**Comment by peter-gribanov** [comment]
Thank you for finding this bug, but the solution you suggested is not fully working. It is necessary to rework the logic for determining the type of interval.

**Comment by simPod** [comment]
After some considerations I decided to add an `eq` case so the complexity of comparison logic does not grow too high.

File: src/IntervalComparator.php [hunk 3]
```diff
@@ -140,26 +140,36 @@ public function cover(ComparableIntervalInterface $interval)
...
+            }
+        } elseif ($this->interval->startPoint()->lt($interval->startPoint())) {
             $start = $this->interval->startPoint();
             if ($this->interval->type()->startExcluded()) {
                 $type |= IntervalType::TYPE_START_EXCLUDED;
             }
         } else {
             $start = $interval->startPoint();
-            if ($interval->type()->startExcluded()) {
+            if ($this->interval->startPoint()->neq($start) && $interval->type()->startExcluded()) {
```
Comment (peter-gribanov) [inline src/IntervalComparator.php]: `$this->interval->startPoint()->neq($start)` always return `false` here.

File: src/IntervalComparator.php [hunk 4]
```diff
@@ -140,26 +140,36 @@ public function cover(ComparableIntervalInterface $interval)
...
+            }
+        } elseif ($this->interval->endPoint()->gt($interval->endPoint())) {
             $end = $this->interval->endPoint();
             if ($this->interval->type()->endExcluded()) {
                 $type |= IntervalType::TYPE_END_EXCLUDED;
             }
         } else {
             $end = $interval->endPoint();
-            if ($interval->type()->endExcluded()) {
+            if ($this->interval->endPoint()->neq($end) && $interval->type()->endExcluded()) {
```
Comment (peter-gribanov) [inline src/IntervalComparator.php]: `$this->interval->endPoint()->neq($end)` always return `false` here.

**Review by peter-gribanov — CHANGES_REQUESTED** [review]

File: src/IntervalComparator.php (hunk 3 above)
Comment (simPod) [reply src/IntervalComparator.php]: This is big truth!

I see, ruled that out above.

**Review by peter-gribanov — APPROVED** [review]

**Comment by peter-gribanov** [comment]
Good job! Thanks :+1:

**Pull request merged**